File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void buildPyramid (int n);

int
main (void)
{
    int rows;
    do
        {
            rows = get_int ("How many rows do you want me to build? ");
        }
    while (rows < 1);

    buildPyramid (rows);
    // return 0;
}

void
buildPyramid (int n)
{
    for (int i = 0; i < n; i++)
        {
            int spaceCount = n - i - 1;
            while (spaceCount >= 1)
                {
                    spaceCount -= 1;
                    printf (" ");
                }

            for (int j = 0; j < i + 1; j++)
                {
                    printf ("#");
                }

            printf ("\n");
        }
}

File: sentimental-mario-less/mario.py
def main() -> None:
    count: int = 0

    # while count < 1 or count > 8:
    while not (0 < count < 9):
        try:
            count = int(input("How many rows do you want me to build? "))
            if count < 1 or count > 8:
                print("Please provide a positive integer between 1 and 8.")
        except ValueError:
            print("Not a number!")

    build_pyramid(count)


def build_pyramid(n: int) -> None:
    for i in range(n):
        print(" " * (n - i - 1), end="")
        print("#" * (i + 1))


main()
